lib/src/date-input/Calendar.tsx

```diff
@@ -0,0 +1,495 @@
+import dayjs, { UnitType, Dayjs } from "dayjs";
+import React, { useState, useMemo, useEffect, useRef } from "react";
+import styled from "styled-components";
+import { CalendarPropsType } from "./types";
+
+const weekDays = ["Mo", "Tu", "We", "Th", "Fr", "Sa", "Su"];
+const months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
+
+const firstYear = 1899;
```

Comment:
Why are you creating variables when both `firstYear` and `lastYear` are only used once inside the loop? 🤔

Reply:
If you do it for a better understanding of the code, it makes more sense to have them inside the function that uses them.